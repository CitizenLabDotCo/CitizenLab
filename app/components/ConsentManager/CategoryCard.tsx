import React, { Fragment, FormEvent } from 'react';
import styled from 'styled-components';
import { colors, fontSizes, media } from 'utils/styleUtils';
import { transparentize, hideVisually } from 'polished';
import messages from './messages';
import { FormattedMessage } from 'utils/cl-intl';
import { CustomRadio, Checked } from 'components/UI/Radio';
import { IDestination } from './';

const Container = styled.div`
  display: flex;
  padding: 20px;
  border-radius: ${(props: any) => props.theme.borderRadius};
  background-color: ${transparentize(.95, colors.adminTextColor)};
  border: 1px solid ${colors.separation};
  margin-bottom: 10px;

  &:last-child {
    margin: 0;
  }

  ${media.smallerThanMaxTablet`
    flex-wrap: wrap;
  `}
`;

const TextContainer = styled.div`
  padding: 0 20px;
  overflow-wrap: break-word;
  word-wrap: break-word;
  word-break: break-word;

  p {
    color: ${colors.label};
    overflow-wrap: break-word;
    word-wrap: break-word;
    word-break: break-word;
  }

  ${media.smallerThanMaxTablet`
    padding: 0;
  `}
`;

const InputContainer = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
`;

const StyledInput = styled.input`
  ${hideVisually()}
`;

const Separator = styled.span`
  color: ${colors.label};
  font-weight: 400;
  font-size: ${fontSizes.base}px;
  line-height: 19px;
`;

const SSpan = styled.span`
  padding-left: 10px;
  padding-right: 10px;

  ${media.smallerThanMaxTablet`
    padding-left: 8px;
    padding-right: 8px;
`}
`;

const StyledLabel = styled.label`
  display: flex;
  flex-direction: row;
  align-items: center;
  flex-wrap: nowrap;
  font-size: ${fontSizes.base}px;
  color: ${colors.label};
  cursor: pointer;
  font-weight: 400;
  margin-bottom: 5px;

  input {
    cursor: pointer;
  }

  &:focus-within, &:hover {
    color: black;

    div {
      border-color: black;
    }
  }
`;

const Tools = styled.span`
  font-weight: 500;
`;

interface Props {
  category: string;
  destinations: IDestination[];
  checked: boolean;
  handleChange: (e: FormEvent<HTMLInputElement>) => (void);
}

const CategoryCard = ({ category, destinations, checked, handleChange }: Props) => (
  <Container className="e2e-category">
    <TextContainer>
      <FormattedMessage
        id={`${category}-label`}
        tagName="h2"
        {...messages[category]}
      />
      <InputContainer role="radiogroup">
        <StyledLabel htmlFor={`${category}-radio-true`}>
          <StyledInput
            type="radio"
            name={category}
            id={`${category}-radio-true`}
            value="true"
            checked={checked === true}
            aria-checked={checked === true}
            onChange={handleChange}
            required
          />
          <CustomRadio
            className={`${checked === true ? 'checked' : ''}`}
          >
            {checked === true && <Checked color="#49B47D" />}
          </CustomRadio>
          <FormattedMessage {...messages.allow} />
        </StyledLabel>
        <StyledLabel htmlFor={`${category}-radio-false`}>
          <StyledInput
            type="radio"
            name={category}
            id={`${category}-radio-false`}
            value="false"
            checked={checked === false}
            aria-checked={checked === false}
            onChange={handleChange}
            required
          />
          <CustomRadio
            className={`${checked === false ? 'checked' : ''}`}
          >
            {checked === false && <Checked color="#49B47D" />}
          </CustomRadio>
          <FormattedMessage {...messages.disallow} />
        </StyledLabel>
      </InputContainer>
      <FormattedMessage
        tagName="p"
        {...messages[`${category}Purpose`]}
      />
      <p>
        <Tools><FormattedMessage {...messages.tools} />:</Tools>
        {destinations.map((d, index) => (
          <Fragment key={d.id}>
            {index !== 0 &&
              <Separator>â€¢</Separator>
            }
            <SSpan>{d.name}</SSpan>
          </Fragment>
        ))}
      </p>
    </TextContainer>
  </Container>
);

export default CategoryCard;
