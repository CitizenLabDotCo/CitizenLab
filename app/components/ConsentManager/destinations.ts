import { isFeatureActive } from 'components/FeatureFlag';
import { ITenantData } from 'services/tenant';
import { IUserData } from 'services/users';

export interface IDestinationMap {
  google_analytics: 'google_analytics';
}

export type IDestination = IDestinationMap[keyof IDestinationMap];
export const CATEGORIES = ['analytics', 'advertising', 'functional'] as const;
export type TCategory = typeof CATEGORIES[number];

const destinationConfigs: IDestinationConfig[] = [];

export interface IDestinationConfig {
  key: IDestination;
  category: TCategory;
  feature_flag?: string;
  hasPermission?: (user?: IUserData) => boolean;
}

export const getDestinationConfigs = () => {
  return destinationConfigs;
};

export const allCategories = () => {
  return CATEGORIES;
};

export const registerDestination = (destinationConfig: IDestinationConfig) => {
  destinationConfigs.push(destinationConfig);
};

export const isDestinationActive = (
  config: IDestinationConfig,
  tenant: ITenantData,
  user?: IUserData | null
): boolean => {
  if (config?.feature_flag && !isFeatureActive(config.feature_flag, tenant)) {
    return false;
  }

  if (config?.hasPermission && !config?.hasPermission(user || undefined)) {
    return false;
  }

  return true;
};
