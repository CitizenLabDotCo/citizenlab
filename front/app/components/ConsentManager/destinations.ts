import { IAppConfigurationData } from 'services/appConfiguration';
import { IUserData } from 'services/users';

export interface IDestinationMap {}

export type IDestination = IDestinationMap[keyof IDestinationMap];
export const CATEGORIES = ['analytics', 'advertising', 'functional'] as const;
export type TCategory = typeof CATEGORIES[number];

const destinationConfigs: IDestinationConfig[] = [];

export interface IConsentManagerFeatureMap {}
export type TConsentManagerFeature =
  IConsentManagerFeatureMap[keyof IConsentManagerFeatureMap];

export interface IDestinationConfig {
  /** A unique key, used to name the destination in the stored cookie */
  key: IDestination;
  /** Destinations are grouped in categories. Under which category should it be listed? */
  category: TCategory | ((tenant: IAppConfigurationData) => TCategory);
  /** The name of the feature flag that should be active for the destination to be functional */
  feature_flag?: TConsentManagerFeature;
  /** Can the destination be active for the given user? */
  hasPermission?: (user?: IUserData) => boolean;
  /** Name of the destination shown in the UI */
  name?: (tenant: IAppConfigurationData) => JSX.Element | string;
}

export const getDestinationConfigs = () => {
  return destinationConfigs;
};

export const getDestinationConfig = (destination: IDestination) => {
  return destinationConfigs.find((d) => d.key === destination);
};

export const allCategories = () => {
  return CATEGORIES;
};

export const registerDestination = (destinationConfig: IDestinationConfig) => {
  destinationConfigs.push(destinationConfig);
};

export const isDestinationActive = (
  config: IDestinationConfig,
  tenant: IAppConfigurationData,
  user?: IUserData | null
): boolean => {
  if (config.feature_flag && !isFeatureActive(config.feature_flag, tenant)) {
    return false;
  }

  if (config.hasPermission && !config.hasPermission(user || undefined)) {
    return false;
  }

  return true;
};

function isFeatureActive(
  featureName: string,
  appConfig: IAppConfigurationData
) {
  return (
    appConfig.attributes.settings[featureName]?.allowed &&
    appConfig.attributes.settings[featureName]?.enabled
  );
}
