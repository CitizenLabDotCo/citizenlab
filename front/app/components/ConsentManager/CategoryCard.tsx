import React, { Fragment, FormEvent } from 'react';
import styled from 'styled-components';
import { colors, fontSizes, media } from 'utils/styleUtils';
import { transparentize } from 'polished';
import messages from './messages';
import { FormattedMessage } from 'utils/cl-intl';
import { Radio } from '@citizenlab/cl2-component-library';
import { getDestinationConfig, IDestination, TCategory } from './destinations';
import useAppConfiguration from 'hooks/useAppConfiguration';
import { isNilOrError } from 'utils/helperUtils';
import { IAppConfigurationData } from 'services/appConfiguration';

const Container = styled.div`
  display: flex;
  padding: 20px;
  border-radius: ${(props) => props.theme.borderRadius};
  background-color: ${transparentize(0.95, colors.primary)};
  border: 1px solid ${colors.divider};
  margin-bottom: 10px;

  &:last-child {
    margin: 0;
  }

  ${media.tablet`
    flex-wrap: wrap;
  `}
`;

const TextContainer = styled.div`
  padding: 0 20px;
  overflow-wrap: break-word;
  word-wrap: break-word;
  word-break: break-word;

  p {
    color: ${colors.textSecondary};
    overflow-wrap: break-word;
    word-wrap: break-word;
    word-break: break-word;
  }

  ${media.tablet`
    padding: 0;
  `}
`;

const StyledFieldset = styled.fieldset`
  display: flex;
  flex-direction: column;
  padding: 0;
  margin: 0;
  margin-bottom: 10px;
  border: none;
`;

const Separator = styled.span`
  color: ${colors.textSecondary};
  font-weight: 400;
  font-size: ${fontSizes.base}px;
  line-height: 19px;
`;

const SSpan = styled.span`
  padding-left: 10px;
  padding-right: 10px;

  ${media.tablet`
    padding-left: 8px;
    padding-right: 8px;
`}
`;

const Tools = styled.span`
  font-weight: 500;
`;

type TConsentCategory = TCategory | 'required';

interface Props {
  category: TConsentCategory;
  destinations: IDestination[];
  checked: boolean;
  disableUncheck?: boolean;
  handleChange: (
    category: TConsentCategory,
    value: boolean
  ) => (e: FormEvent<HTMLInputElement>) => void;
}

const DestinationName = ({
  tenant,
  destination,
}: {
  tenant: IAppConfigurationData | null;
  destination: IDestination;
}) => {
  const config = getDestinationConfig(destination);
  if (config?.name && tenant) {
    return <>{config.name(tenant)}</>;
  } else if (config) {
    return <>{config.key}</>;
  } else {
    return null;
  }
};

const CategoryCard = ({
  category,
  destinations,
  checked,
  handleChange,
  disableUncheck,
}: Props) => {
  const appConfig = useAppConfiguration();

  return (
    <Container className="e2e-category">
      <TextContainer>
        <FormattedMessage
          tagName="h2"
          {...{
            functional: messages.functional,
            advertising: messages.advertising,
            analytics: messages.analytics,
            required: messages.required,
          }[category]}
        />
        <StyledFieldset>
          <Radio
            onChange={handleChange(category, true)}
            currentValue={checked}
            value={true}
            name={category}
            id={`${category}-radio-true`}
            label={<FormattedMessage {...messages.allow} />}
            isRequired
          />
          <Radio
            onChange={handleChange(category, false)}
            currentValue={checked}
            value={false}
            name={category}
            id={`${category}-radio-false`}
            label={<FormattedMessage {...messages.disallow} />}
            isRequired
            disabled={disableUncheck}
          />
        </StyledFieldset>
        <FormattedMessage
          tagName="p"
          {...{
            functional: messages.functionalPurpose,
            advertising: messages.advertisingPurpose,
            analytics: messages.analyticsPurpose,
            required: messages.requiredPurpose,
          }[category]}
        />
        {destinations.length > 0 && (
          <p>
            <Tools>
              <FormattedMessage {...messages.tools} />:
            </Tools>
            {destinations.map((d, index) => (
              <Fragment key={d}>
                {index !== 0 && <Separator>â€¢</Separator>}
                <SSpan>
                  <DestinationName
                    tenant={!isNilOrError(appConfig) ? appConfig : null}
                    destination={d}
                  />
                </SSpan>
              </Fragment>
            ))}
          </p>
        )}
      </TextContainer>
    </Container>
  );
};

export default CategoryCard;
