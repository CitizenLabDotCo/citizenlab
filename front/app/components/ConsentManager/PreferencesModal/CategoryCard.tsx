import React, { Fragment } from 'react';

import {
  Radio,
  Title,
  colors,
  fontSizes,
  media,
} from '@citizenlab/cl2-component-library';
import { transparentize } from 'polished';
import styled from 'styled-components';

import { IAppConfigurationData } from 'api/app_configuration/types';
import useAppConfiguration from 'api/app_configuration/useAppConfiguration';

import { FormattedMessage } from 'utils/cl-intl';
import { isNilOrError } from 'utils/helperUtils';

import { getDestinationConfig, IDestination, TCategory } from '../destinations';
import messages from '../messages';

const Container = styled.div`
  display: flex;
  padding: 20px;
  border-radius: ${(props) => props.theme.borderRadius};
  background-color: ${transparentize(0.95, colors.primary)};
  border: 1px solid ${colors.divider};
  margin-bottom: 10px;

  &:last-child {
    margin: 0;
  }

  ${media.tablet`
    flex-wrap: wrap;
  `}
`;

const TextContainer = styled.div`
  padding: 0 20px;
  overflow-wrap: break-word;
  word-wrap: break-word;
  word-break: break-word;

  p {
    color: ${colors.textSecondary};
    overflow-wrap: break-word;
    word-wrap: break-word;
    word-break: break-word;
  }

  ${media.tablet`
    padding: 0;
  `}
`;

const StyledFieldset = styled.fieldset`
  display: flex;
  flex-direction: column;
  padding: 0;
  margin: 0;
  margin-bottom: 10px;
  border: none;
`;

const Separator = styled.span`
  color: ${colors.textSecondary};
  font-weight: 400;
  font-size: ${fontSizes.base}px;
  line-height: 19px;
`;

const SSpan = styled.span`
  padding-left: 10px;
  padding-right: 10px;

  ${media.tablet`
    padding-left: 8px;
    padding-right: 8px;
`}
`;

const Tools = styled.span`
  font-weight: 500;
`;

type TConsentCategory = TCategory | 'required';

interface Props {
  category: TConsentCategory;
  destinations: IDestination[];
  checked: boolean;
  disableUncheck?: boolean;
  onChange: (category: TCategory, value: boolean) => void;
}

const DestinationName = ({
  tenant,
  destination,
}: {
  tenant: IAppConfigurationData | null;
  destination: IDestination;
}) => {
  const config = getDestinationConfig(destination);
  if (config?.name && tenant) {
    return <>{config.name(tenant)}</>;
  } else if (config) {
    return <>{config.key}</>;
  } else {
    return null;
  }
};

const RADIO_IDS_TRUE: Record<TCategory, string> = {
  advertising: 'advertising-radio-true',
  analytics: 'analytics-radio-true',
  functional: 'functional-radio-true',
};

const RADIO_IDS_FALSE: Record<TCategory, string> = {
  advertising: 'advertising-radio-false',
  analytics: 'analytics-radio-false',
  functional: 'functional-radio-false',
};

const CategoryCard = ({
  category,
  destinations,
  checked,
  onChange,
  disableUncheck,
}: Props) => {
  const { data: appConfig } = useAppConfiguration();

  const handleChange = (category: TCategory, value: boolean) => () => {
    onChange(category, value);
  };

  return (
    <Container className="e2e-category">
      <TextContainer>
        <Title styleVariant="h4" as="h2">
          <FormattedMessage
            {...{
              functional: messages.functional,
              advertising: messages.advertising,
              analytics: messages.analytics,
              required: messages.required,
            }[category]}
          />
        </Title>
        <StyledFieldset>
          <Radio
            onChange={
              category === 'required' ? undefined : handleChange(category, true)
            }
            currentValue={checked}
            value={true}
            name={category}
            id={RADIO_IDS_TRUE[category]}
            label={<FormattedMessage {...messages.allow} />}
            isRequired
          />
          <Radio
            onChange={
              category === 'required'
                ? undefined
                : handleChange(category, false)
            }
            currentValue={checked}
            value={false}
            name={category}
            id={RADIO_IDS_FALSE[category]}
            label={<FormattedMessage {...messages.disallow} />}
            isRequired
            disabled={disableUncheck}
          />
        </StyledFieldset>
        <FormattedMessage
          tagName="p"
          {...{
            functional: messages.functionalPurpose,
            advertising: messages.advertisingPurpose,
            analytics: messages.analyticsPurpose,
            required: messages.requiredPurpose,
          }[category]}
        />
        {destinations.length > 0 && (
          <p>
            <Tools>
              <FormattedMessage {...messages.tools} />:
            </Tools>
            {destinations.map((d, index) => (
              <Fragment key={d}>
                {index !== 0 && <Separator>â€¢</Separator>}
                <SSpan>
                  <DestinationName
                    tenant={!isNilOrError(appConfig) ? appConfig.data : null}
                    destination={d}
                  />
                </SSpan>
              </Fragment>
            ))}
          </p>
        )}
      </TextContainer>
    </Container>
  );
};

export default CategoryCard;
