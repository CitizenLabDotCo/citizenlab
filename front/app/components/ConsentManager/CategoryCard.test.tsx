// @ts-nocheck
// libraries
import React from 'react';
import { shallow } from 'enzyme';

import 'jest-styled-components';

// component to test
import CategoryCard from './CategoryCard';
import { registerDestination } from './destinations';

// mock utilities
jest.mock('services/appConfiguration');
jest.mock('utils/cl-intl');
jest.mock('modules', () => ({ streamsToReset: [] }));

describe('<CategoryCard />', () => {
  const category = 'analytics';

  const destinations = ['google_analytics'];
  registerDestination({
    key: 'google_analytics',
    category: 'analytics',
    name: () => 'Google Analytics',
  });
  const handleChange = jest.fn();

  beforeEach(() => {});

  it('renders correctly when null', () => {
    const wrapper = shallow(
      <CategoryCard
        key={category}
        category={category}
        destinations={destinations}
        checked={null}
        handleChange={handleChange}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly when allowed', () => {
    const wrapper = shallow(
      <CategoryCard
        key={category}
        category={category}
        destinations={destinations}
        checked={true}
        handleChange={handleChange}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly when disallowed', () => {
    const wrapper = shallow(
      <CategoryCard
        key={category}
        category={category}
        destinations={destinations}
        checked={false}
        handleChange={handleChange}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('lets you change your preference', () => {
    const mockOnChange = jest.fn();
    handleChange.mockImplementation(() => mockOnChange);

    const wrapper = shallow(
      <CategoryCard
        key={category}
        category={category}
        destinations={destinations}
        checked={null}
        handleChange={handleChange}
      />
    );
    // when you render the component, handle change is called to calculate the onchange handlers.
    expect(handleChange).toHaveBeenCalledWith(category, false);
    expect(handleChange).toHaveBeenCalledWith(category, true);

    // when you change the value these onchange handlers should be called
    wrapper.find(`#${category}-radio-false`).simulate('change');
    expect(mockOnChange).toHaveBeenCalledTimes(1);

    wrapper.find(`#${category}-radio-true`).simulate('change');
    expect(mockOnChange).toHaveBeenCalledTimes(2);
  });
});
