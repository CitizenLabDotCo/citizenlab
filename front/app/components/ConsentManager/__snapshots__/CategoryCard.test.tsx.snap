// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryCard /> renders correctly when allowed 1`] = `
<CategoryCard__Container
  className="e2e-category"
>
  <CategoryCard__TextContainer>
    <MemoizedFormattedMessage
      defaultMessage="Marketing and analytics"
      id="app.components.ConsentManager.Modal.PreferencesDialog.analytics"
      tagName="h2"
    />
    <CategoryCard__StyledFieldset>
      <Radio
        currentValue={true}
        id="analytics-radio-true"
        isRequired={true}
        label={
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Allow"
            id="app.components.ConsentManager.Modal.PreferencesDialog.allow"
          />
        }
        name="analytics"
        value={true}
      />
      <Radio
        currentValue={true}
        id="analytics-radio-false"
        isRequired={true}
        label={
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Disallow"
            id="app.components.ConsentManager.Modal.PreferencesDialog.disallow"
          />
        }
        name="analytics"
        value={false}
      />
    </CategoryCard__StyledFieldset>
    <MemoizedFormattedMessage
      defaultMessage="We use this tracking to understand better how you use the platform in order to learn and improve your navigation. This information is only used in mass analytics, and in no way to track individuals."
      id="app.components.ConsentManager.Modal.PreferencesDialog.analyticsPurpose"
      tagName="p"
    />
    <p>
      <CategoryCard__Tools>
        <MemoizedFormattedMessage
          defaultMessage="Tools"
          id="app.components.ConsentManager.Modal.PreferencesDialog.tools"
        />
        :
      </CategoryCard__Tools>
      <CategoryCard__SSpan>
        <DestinationName
          destination="google_analytics"
          tenant={null}
        />
      </CategoryCard__SSpan>
    </p>
  </CategoryCard__TextContainer>
</CategoryCard__Container>
`;

exports[`<CategoryCard /> renders correctly when disallowed 1`] = `
<CategoryCard__Container
  className="e2e-category"
>
  <CategoryCard__TextContainer>
    <MemoizedFormattedMessage
      defaultMessage="Marketing and analytics"
      id="app.components.ConsentManager.Modal.PreferencesDialog.analytics"
      tagName="h2"
    />
    <CategoryCard__StyledFieldset>
      <Radio
        currentValue={false}
        id="analytics-radio-true"
        isRequired={true}
        label={
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Allow"
            id="app.components.ConsentManager.Modal.PreferencesDialog.allow"
          />
        }
        name="analytics"
        value={true}
      />
      <Radio
        currentValue={false}
        id="analytics-radio-false"
        isRequired={true}
        label={
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Disallow"
            id="app.components.ConsentManager.Modal.PreferencesDialog.disallow"
          />
        }
        name="analytics"
        value={false}
      />
    </CategoryCard__StyledFieldset>
    <MemoizedFormattedMessage
      defaultMessage="We use this tracking to understand better how you use the platform in order to learn and improve your navigation. This information is only used in mass analytics, and in no way to track individuals."
      id="app.components.ConsentManager.Modal.PreferencesDialog.analyticsPurpose"
      tagName="p"
    />
    <p>
      <CategoryCard__Tools>
        <MemoizedFormattedMessage
          defaultMessage="Tools"
          id="app.components.ConsentManager.Modal.PreferencesDialog.tools"
        />
        :
      </CategoryCard__Tools>
      <CategoryCard__SSpan>
        <DestinationName
          destination="google_analytics"
          tenant={null}
        />
      </CategoryCard__SSpan>
    </p>
  </CategoryCard__TextContainer>
</CategoryCard__Container>
`;

exports[`<CategoryCard /> renders correctly when null 1`] = `
<CategoryCard__Container
  className="e2e-category"
>
  <CategoryCard__TextContainer>
    <MemoizedFormattedMessage
      defaultMessage="Marketing and analytics"
      id="app.components.ConsentManager.Modal.PreferencesDialog.analytics"
      tagName="h2"
    />
    <CategoryCard__StyledFieldset>
      <Radio
        currentValue={null}
        id="analytics-radio-true"
        isRequired={true}
        label={
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Allow"
            id="app.components.ConsentManager.Modal.PreferencesDialog.allow"
          />
        }
        name="analytics"
        value={true}
      />
      <Radio
        currentValue={null}
        id="analytics-radio-false"
        isRequired={true}
        label={
          <Memo(MemoizedFormattedMessage)
            defaultMessage="Disallow"
            id="app.components.ConsentManager.Modal.PreferencesDialog.disallow"
          />
        }
        name="analytics"
        value={false}
      />
    </CategoryCard__StyledFieldset>
    <MemoizedFormattedMessage
      defaultMessage="We use this tracking to understand better how you use the platform in order to learn and improve your navigation. This information is only used in mass analytics, and in no way to track individuals."
      id="app.components.ConsentManager.Modal.PreferencesDialog.analyticsPurpose"
      tagName="p"
    />
    <p>
      <CategoryCard__Tools>
        <MemoizedFormattedMessage
          defaultMessage="Tools"
          id="app.components.ConsentManager.Modal.PreferencesDialog.tools"
        />
        :
      </CategoryCard__Tools>
      <CategoryCard__SSpan>
        <DestinationName
          destination="google_analytics"
          tenant={null}
        />
      </CategoryCard__SSpan>
    </p>
  </CategoryCard__TextContainer>
</CategoryCard__Container>
`;
